### PR TITLE
Add telemetry for `SourceBuffer.appendBuffer`

### PR DESCRIPTION
**Pull request by at-ninja**
Modifies the SourceBufferMetrics class to not be completely stubbed out in gold builds, and use it to send UMA histograms for the time spent in `SourceBuffer::PrepareAppend` and
`SourceBufferAppendAlgorithm::Process`.

Added basic unit tests for the `SourceBufferMetrics` class, to assert that the new code reports telemetry against the correct histograms.

b/329439818

Change-Id: I7921434b02c689b645ab2a60464eee166d829135

### CODE REVIEW

File: cobalt/dom/source_buffer_metrics.h
Comment (sideb0ard) [inline cobalt/dom/source_buffer_metrics.h:39]: lets add the SourceBufferMetricsAction enum to StartTracking too, to be consistent with EndTracking. Lets add a DCHECK or similar to ensure the EndTracking matches the correct StartTracking

**Review by sideb0ard — CHANGES_REQUESTED** [review]
Looks great! Added some minor comments.

File: tools/metrics/histograms/metadata/cobalt/histograms.xml
Comment (joeltine) [reply tools/metrics/histograms/metadata/cobalt/histograms.xml:34]: Correct. Sorry about that. Just haven't implemented this yet.

File: cobalt/dom/source_buffer_metrics.cc
Comment (joeltine) [inline cobalt/dom/source_buffer_metrics.cc:121]: Were you able to test these histograms locally? How was your bucket distribution? I wonder about your min/max ranges here and your bucket size (50). 1 microsecond to 1 second is a delta of 999,999. This is a very wide range... is 50 buckets too coarse? Can you tighten your interval here reasonably and also perhaps up the buckets to 100?

**Review by sideb0ard — APPROVED** [review]
thanks for the changes, lgtm, pending Joel's approval of histogram buckets.

File: tools/metrics/histograms/metadata/cobalt/histograms.xml
Comment (joeltine) [inline tools/metrics/histograms/metadata/cobalt/histograms.xml:155]: aren't your units microseconds?

File: cobalt/dom/source_buffer_metrics.cc
Comment (joeltine) [reply cobalt/dom/source_buffer_metrics.cc]: Looks good.

File: tools/metrics/histograms/metadata/cobalt/histograms.xml
Comment (at-ninja) [reply tools/metrics/histograms/metadata/cobalt/histograms.xml]: Fixed.

**Review by joeltine — APPROVED** [review]

**Pull request merged**
